src/_pytest/recwarn.py

```diff
@@ -108,6 +108,7 @@ def warns(  # noqa: F811
     expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]] = Warning,
     *args: Any,
     match: Optional[Union[str, Pattern[str]]] = None,
+    re_emit: bool = False,
```

Comment:
I think this new feature should be unconditionally enabled, which will simplify it for both users and maintainers.